operator/pkg/gateway-api/grpcroute.go

```diff
@@ -85,34 +96,67 @@ func getParentGatewayForGRPCRoute(rawObj client.Object) []string {
 	return gateways
 }
 
-func getBackendServiceForGRPCRoute(rawObj client.Object) []string {
+func (r *grpcRouteReconciler) getBackendServiceForGRPCRoute(rawObj client.Object) []string {
 	route, ok := rawObj.(*gatewayv1alpha2.GRPCRoute)
 	if !ok {
 		return nil
 	}
 	var backendServices []string
 	for _, rule := range route.Spec.Rules {
 		for _, backend := range rule.BackendRefs {
-			if !helpers.IsService(backend.BackendObjectReference) {
-				continue
+			namespace := helpers.NamespaceDerefOr(backend.Namespace, route.Namespace)
+			backendServiceName, err := helpers.GetBackendServiceName(r.Client, namespace, backend.BackendObjectReference)
+			if err != nil {
+				log.WithFields(logrus.Fields{
+					logfields.Controller: "grpcRoute",
+					logfields.Resource:   client.ObjectKeyFromObject(rawObj),
+				}).WithError(err).Error("Failed to get backend service name")
```

Comment:
in case of an error, we need to `continue` without adding the service to the slice